botwo/botwo.py

```diff
@@ -0,0 +1,188 @@
+import fnmatch
+
+SPECIAL_METHODS = {"get_paginator", "delete_objects"}
+COPY_METHODS = {"copy", "copy_object", "copy_upload_part"}
+
+
+def client(client_mapping, config):
+    botwo = Botwo()
+    return botwo.create_client_class(client_mapping, config)
+
+
+class Paginatwo(object):
+
+    def __init__(self, mapping, kwargs, client1, client2, client1_name, client2_name):
+        self.mapping = mapping
+        self.client1_name = client1_name
+        self.client2_name = client2_name
+        self.paginator1 = client1.get_paginator(kwargs["operation_name"])
+        self.paginator2 = client2.get_paginator(kwargs["operation_name"])
+
+    def paginate(self, **kwargs):
+        paginator_to_call = self.paginator1
+        for map in self.mapping:
+            if fnmatch.fnmatch(kwargs["Bucket"], map["bucket_name"]):
+                if map.get("client"):
+                    if map.get("client") == self.client2_name:
+                        paginator_to_call = self.paginator2
+                    elif map.get("client") == self.client1_name:
+                        paginator_to_call = self.paginator1
+                    else:
+                        raise ValueError("Invalid client name " + map.get("client"))
+                else:
+                    raise ValueError("client isn't provided")
+
+                if map.get("mapped_bucket_name"):
+                    kwargs["Bucket"] = map.get("mapped_bucket_name")
+                if kwargs.get("Prefix"):
+                    if map.get("key_prefix"):
+                        kwargs["Prefix"] = map.get("keu_prefix") + kwargs["Prefix"]
```

Comment:
How was this tested?  I believe there is an error on this line.
```suggestion
                        kwargs["Prefix"] = map.get("key_prefix") + kwargs["Prefix"]
```

Reply:
Done